Bug: Partial struct initialization creates struct with missing fields

## Description

When using partial struct literal initialization (e.g., `Point{x: 5}` without specifying `y`), the struct is created but the unspecified fields don't exist at all. Accessing them results in E4003 "field not found" error. This is inconsistent with `new(StructType)` which correctly zero-initializes all fields.

## Steps to Reproduce

```ez
import @std
using std

const Point struct {
    x int
    y int
}

do main() {
    // This works - all fields zero-initialized
    temp p1 = new(Point)
    println("new(Point): x=${p1.x}, y=${p1.y}")  // OK: x=0, y=0

    // This fails - y field doesn't exist
    temp p2 Point = Point{x: 5}
    println("Point{x: 5}: x=${p2.x}, y=${p2.y}")  // ERROR: field 'y' not found
}
```

## Expected Behavior

Either:
1. **Option A**: Partial initialization should zero-initialize missing fields (consistent with `new()`)
2. **Option B**: Partial initialization should be a compile-time error requiring all fields to be specified

## Actual Behavior

```
new(Point): x=0, y=0
error[E4003]: field 'y' not found
  --> test.ez:16:16
   |
16 |     println("Point{x: 5}: x=${p2.x}, y=${p2.y}")
   |                ^ field does not exist on type
```

The struct is created with only the specified field `x`, and field `y` is completely missing (not zero-initialized).

## Impact

- **Inconsistent behavior**: `new(Point)` and `Point{...}` behave differently for unspecified fields
- **Confusing error message**: Says "field 'y' not found" which sounds like Point doesn't have a y field
- **Silent partial initialization**: Users might not realize they forgot to initialize a field until runtime

## Environment

- EZ Version: dev (testing against main branch)
- OS: macOS Darwin 24.5.0

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -2820,10 +2820,13 @@ func evalStructValue(node *ast.StructValue, env *Environment) Object {
 		}
 	}
 
-	// Create a new struct with the given fields
+	// Create a new struct with default values for all fields first
 	fields := make(map[string]Object)
+	for fieldName, fieldType := range structDef.Fields {
+		fields[fieldName] = getDefaultValue(fieldType)
+	}
 
-	// Evaluate each field expression
+	// Override with explicitly provided field values
 	for fieldName, fieldExpr := range node.Fields {
 		val := Eval(fieldExpr, env)
 		if isError(val) {

